### PR TITLE
General redesign of ImageVisualizer

### PR DESCRIPTION
**Pull request by robinschmid**
Still in progress, includes the imzML fix and the spectraPlot fix.

Now uses the datasets and has a ImagingPlot that is reusable and allows to set new datasets like the SpectraPlot

### CODE REVIEW

**Comment by robinschmid** [comment]
Ready for now. THe visualizer needs more attention later

File: src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java [hunk 1]
```diff
@@ -607,4 +613,29 @@ public void applyWithNotifyChanges(boolean tempState, boolean afterRunState, Run
       }
     }
   }
+
+  public Marker addDomainMarker(com.google.common.collect.Range<Double> mzRange, Color color,
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java:617]: i'd rename this to valueRange or so, it's not always mz

File: src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java [hunk 2]
```diff
@@ -607,4 +613,29 @@ public void applyWithNotifyChanges(boolean tempState, boolean afterRunState, Run
...
+      float alpha) {
+    return addDomainMarker(mzRange.lowerEndpoint(), mzRange.upperEndpoint(), color, alpha);
+  }
+
+  public Marker addDomainMarker(double precursorMz, Color color, float alpha) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java:622]: same

File: src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java [hunk 3]
```diff
@@ -607,4 +613,29 @@ public void applyWithNotifyChanges(boolean tempState, boolean afterRunState, Run
...
+    final ValueMarker marker = new ValueMarker(precursorMz);
+    marker.setStroke(
+        new BasicStroke(2f, BasicStroke.CAP_BUTT, BasicStroke.JOIN_MITER, 10f, new float[]{7}, 0f));
+    marker.setPaint(color);
+    marker.setAlpha(alpha);
+    getChart().getXYPlot().addDomainMarker(marker, Layer.BACKGROUND);
+    return marker;
+  }
+
+  public Marker addDomainMarker(double lowerMZ, double upperMZ, Color color, float alpha) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/gui/chartbasics/gui/javafx/EChartViewer.java:632]: same

File: src/main/java/io/github/mzmine/modules/visualization/image/ImagingPlot.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    final boolean lockOnAspectRatio = MZmineCore.getConfiguration()
+        .getModuleParameters(FeatureTableFXModule.class)
+        .getParameter(FeatureTableFXParameters.lockImagesToAspectRatio).getValue();
+    chart.getXYPlot().setBackgroundPaint(Color.BLACK);
+
+    setCenter(chart);
+    return chart;
+  }
+
+  public void addDataset(ColoredXYZDataset ds) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/visualization/image/ImagingPlot.java:117]: if this doesnt do anything, dont include it yet

File: src/main/java/io/github/mzmine/modules/visualization/image/RawImageProvider.java [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+
+    logger.info("ImageViewer: Start data point extraction");
+    ScanDataAccess scanAccess = EfficientDataAccess.of(raw, ScanDataType.RAW, scanSelection);
+    int numberOfScans = scanAccess.getNumberOfScans();
+    DoubleArrayList mzs = new DoubleArrayList(numberOfScans);
+    DoubleArrayList intensities = new DoubleArrayList(numberOfScans);
+    List<Scan> scans = new ArrayList<>(numberOfScans);
+    int finished = 0;
+    while (scanAccess.hasNextScan()) {
+      finishedPercentage = finished / numberOfScans;
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/visualization/image/RawImageProvider.java:215]: you are not increasing finished, or?

File: src/main/java/io/github/mzmine/modules/visualization/image/RawImageProvider.java [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+          if (mz < minMz) {
+            continue;
+          }
+
+          // sum intensity
+          double intensity = scanAccess.getIntensityValue(i);
+          sum += intensity;
+          // find best signal
+          if (intensity > maxIntensity) {
+            maxIntensity = intensity;
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/visualization/image/RawImageProvider.java:295]: what is the max intensity used for? we are updating the local var here

File: src/main/java/io/github/mzmine/modules/visualization/chromatogramandspectra/ChromatogramAndSpectraVisualizer.java [hunk 7]
```diff
@@ -423,7 +422,6 @@ private void onChromatogramSelectionChanged(
    */
   private void onSpectrumSelectionChanged(ObservableValue<? extends SpectrumCursorPosition> obs,
       SpectrumCursorPosition old, SpectrumCursorPosition pos) {
-    updateSpectrumDomainMarker(pos);
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/visualization/chromatogramandspectra/ChromatogramAndSpectraVisualizer.java:426]: does this still work then? there's only removed code here

Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/visualization/chromatogramandspectra/ChromatogramAndSpectraVisualizer.java:426]: its all done in the SpectraPlot and updated via bindings. was working for mein the image visualizer and the raw data overview

File: src/main/java/io/github/mzmine/modules/visualization/image/RawImageProvider.java (hunk 5 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/visualization/image/RawImageProvider.java:215]: true

File: src/main/java/io/github/mzmine/modules/visualization/image/RawImageProvider.java (hunk 6 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/visualization/image/RawImageProvider.java:295]: only to find the best mz - as we are using mz in the IonTimeSeries

File: src/main/java/io/github/mzmine/modules/visualization/image/ImagingPlot.java (hunk 4 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/visualization/image/ImagingPlot.java:117]: removed

**Comment by robinschmid** [comment]
All done

**Pull request merged**